### PR TITLE
Added default headers option

### LINKED ISSUE #87: Default headers
Allow the configuration of default headers. All the response will send those default values.
Developers MUST have the chance to override one or more headers in each single action.

Default headers are expressed as a Hash, where keys are the header name and values the header value. The default value SHOULD be an empty Hash.

```ruby
Lotus::Controller.configure do
  default_headers({
    'X-Foo' => 'bar',
    'Content-Security-Policy' => '...'
  })
end
```

### PR DESCRIPTION
**Pull request by AlfonsoUceda**
closes  #87

### CODE REVIEW

File: lib/lotus/action/head.rb [hunk 1]
```diff
@@ -26,6 +26,8 @@ def finish
 
         if _requires_no_body?
           @_body = nil
+        else
+          @headers.merge!(configuration.default_headers)
```
Comment (AlfonsoUceda) [inline lib/lotus/action/head.rb]: I added this else for merging the default headers if the method isn't HEAD or the status code isn't HTTP_STATUSES_WITHOUT_BODY defined above

File: test/configuration_test.rb [hunk 2]
```diff
@@ -270,6 +270,26 @@ def hash
...
+        @configuration.default_headers(headers)
+      end
+
+      it 'returns the value' do
+        @configuration.default_headers.must_equal headers
+      end
+    end
+  end
+
   describe 'duplicate' do
```
Comment (jodosha) [inline test/configuration_test.rb]: Can you please add this to `duplicate` and `reset!` tests? TY!

**Comment by jodosha** [comment]
@AlfonsoUceda This looks really good. I left one comment for extra tests for `Configuration#duplicate` and `#reset`.

Quoting #87 :

> Developers MUST have the chance to override one or more headers in each single action.

This is still missing and it's really important for resource based headers.

Can you please ensure that it works with a test like this?

```ruby
class MyAction
  include Lotus::Action

  # Default header in configuration is "Content-Security-Policy" => "script-src 'self'"
  def call(params)
    # ...
    headers['Content-Security-Policy'] = "script-src 'self' https://api.example.org"
  end
end
```

The header specified in the action should be the output.

File: lib/lotus/action/head.rb [hunk 3]
```diff
@@ -26,6 +45,7 @@ def finish
 
         if _requires_no_body?
           @_body = nil
+          @headers.reject! {|header,_| !ENTITY_HEADERS.include?(header) }
```
Comment (joneslee85) [inline lib/lotus/action/head.rb]: nitpick, add a space after `{`

File: lib/lotus/action/head.rb [hunk 4]
```diff
@@ -14,6 +14,25 @@ module Head
...
+        'Content-Encoding' => true,
+        'Content-Language' => true,
+        'Content-Length'   => true,
+        'Content-Location' => true,
+        'Content-MD5'      => true,
+        'Content-Range'    => true,
+        'Content-Type'     => true,
+        'Expires'          => true,
+        'Last-Modified'    => true,
+        'extension-header' => true
```
Comment (joneslee85) [inline lib/lotus/action/head.rb]: is it supposed to be all lowered case?

File: lib/lotus/controller/configuration.rb [hunk 5]
```diff
@@ -469,6 +469,33 @@ def default_charset(charset = nil)
         end
       end
 
+      # Set default headers for all the responses
```
Comment (joneslee85) [inline lib/lotus/controller/configuration.rb]: /s/the//

File: test/integration/full_stack_test.rb [hunk 6]
```diff
@@ -20,7 +20,7 @@ def app
     head '/head', {}, 'HTTP_ACCEPT' => 'text/html'
 
     last_response.body.must_be_empty
-    last_response.headers['X-Renderable'].must_equal 'false'
+    last_response.headers['X-Renderable'].must_equal nil
```
Comment (joneslee85) [inline test/integration/full_stack_test.rb]: must_be_nil instead?

File: lib/lotus/action/head.rb (hunk 4 above)
Comment (AlfonsoUceda) [inline lib/lotus/action/head.rb]: I don't know, I'll take from here https://github.com/rack/rack/issues/787

**Comment by joneslee85** [comment]
except above comments, :+1: to go

**Comment by jodosha** [comment]
@AlfonsoUceda Thank you, merged :wink:
